src/Controller/AbstractController.php

```diff
@@ -14,6 +14,8 @@ abstract class AbstractController extends Controller
 
     abstract protected function getAttributeRepository() : AttributeRepositoryInterface;
 
+    abstract protected function getAttributeCodeBlacklist() : array;
+
```

Comment:
To avoid breaking changes, we could simply return an empty array as the default here.